src/stores/signaling.ts

```diff
@@ -0,0 +1,343 @@
+					})
+				} else if (sessionIds.length) {
+					// Participant left conversation from all devices
+					store.commit('updateParticipant', {
+						token,
+						attendeeId,
+						updatedData: { inCall: PARTICIPANT.CALL_FLAG.DISCONNECTED, sessionIds: [] },
+					})
+				}
+			}
+
+			// Clean up old sessions
+			for (const session of Object.keys(this.sessions)) {
+				if (!newSessions.has(session)) {
+					this.deleteSignalingSession(session)
+				}
+			}
+
+			return hasUnknownSessions
+		},
+
+		/**
+		 * Update participants joined in store according to data from standalone signaling server
+		 *
+		 * @param token the conversation token;
+		 * @param participants the newly joined participant objects;
+		 * @return {boolean} whether list has unknown sessions mapped to attendees list
+		 */
+		updateParticipantsJoinedFromStandaloneSignaling(token: string, participants: StandaloneSignalingJoinPayload[]): boolean {
+			const attendeeUsers = store.getters.participantsList(token) as Participant[]
+			const attendeeUsersToUpdate: Record<string, { displayName?: string, sessionIds: string[] }> = {}
+			let hasUnknownSessions = false
+
+			for (const participant of participants) {
+				const session = this.findSignalingSession(token, 'ext-join', participant)
+				const attendeeId = session?.attendeeId
+
+				if (!attendeeId) {
+					hasUnknownSessions = true
+					continue
+				}
+
+				const attendee = store.getters.getParticipant(token, attendeeId)
+
+				if (!attendeeUsersToUpdate[attendeeId]) {
+					attendeeUsersToUpdate[attendeeId] = { sessionIds: [...attendee.sessionIds] }
+				}
+				if (participant.user.displayname) {
+					attendeeUsersToUpdate[attendeeId].displayName = participant.user.displayname
+				}
+				// Participant might join from several devices
+				if (!attendeeUsersToUpdate[attendeeId].sessionIds.includes(participant.roomsessionid)) {
+					attendeeUsersToUpdate[attendeeId].sessionIds.push(participant.roomsessionid)
+				}
+			}
+
+			for (const [attendeeId, updatedData] of Object.entries(attendeeUsersToUpdate)) {
+				store.commit('updateParticipant', {
+					token,
+					attendeeId: +attendeeId,
+					updatedData,
+				})
+			}
+
+			return hasUnknownSessions
+		},
+
+		/**
+		 * Update participants left in store according to data from standalone signaling server
+		 *
+		 * @param signalingSessionIds disconnected signaling sessions;
+		 */
+		updateParticipantsLeftFromStandaloneSignaling(signalingSessionIds: string[]) {
+			for (const signalingSessionId of signalingSessionIds) {
+				const session = this.getSignalingSession(signalingSessionId)
+				if (!session) {
+					continue
+				}
+				this.deleteSignalingSession(signalingSessionId)
+
+				const { token, attendeeId, sessionId } = session
+				const attendee = store.getters.getParticipant(token, attendeeId)
+				const updatedData : { sessionIds: [], inCall?: number } = {
+					sessionIds: attendee.sessionIds.filter((id: string) => id !== sessionId)
```

Comment:
When actor leaves conversation
```suggestion
					sessionIds: attendee?.sessionIds.filter((id: string) => id !== sessionId)
```